### ISSUE TITLE
EVENT_UPLOAD_COMPLETE wrong payload when using GCS

### ISSUE THREAD

**Issue by tkrugg** [tus/tus-node-server#129]
It seems when using GCS, the payload of the EVENT_UPLOAD_COMPLETE event is not as stated in the README.md
```js
{
   file: {
        id: '7b26bf4d22cf7198d3b3706bf0379794',
        upload_length: '41767441',
        upload_metadata: 'filename NDFfbWIubXA0'
    }
}
```
instead, file is actually an instance of the [File class](https://github.com/googleapis/nodejs-storage/blob/master/src/file.js#L98) as you can tell by the code:
https://github.com/tus/tus-node-server/blob/master/lib/stores/GCSDataStore.js#L160

**Comment by Acconut** [comment]
Thanks for brining this up. It seems as if that's also the case for other events, such as https://github.com/tus/tus-node-server/blob/master/lib/stores/GCSDataStore.js#L111

Would you mind sending a pull request to fix this issue?

**Comment by tkrugg** [comment]
Sorry for the late reply. Sure, will do.

**Comment by tkrugg** [comment]
Hi @Acconut 
> It seems as if that's also the case for other events, such as https://github.com/tus/tus-node-server/blob/master/lib/stores/GCSDataStore.js#L111
 
Looking at it again it seems the payload of `EVENT_FILE_CREATED` is consistently an instance of [models/File](https://github.com/tus/tus-node-server/blob/master/lib/models/File.js#L10) for DataStore, GCSDataStore and S3Store.

The appear to be inconsistant for EVENT_UPLOAD_COMPLETE however
- [S3Store](https://github.com/tus/tus-node-server/blob/871bc2c3f6fe95d48bc5a6e39e26f5760e0e4904/lib/stores/S3Store.js#L284) conforms to the doc
- [FileStore](https://github.com/tus/tus-node-server/blob/cd9d6cb79f6c05abdf74b89a02f5d792a5189b88/lib/stores/FileStore.js#L127) and  [GCSDataStore](https://github.com/tus/tus-node-server/blob/595514701755ca624519d70650cc7490c7ee4faf/lib/stores/GCSDataStore.js#L134) use a different payload.file

I'm not sure on whether we should clarify the docs or fix this inconsistency

**Comment by Acconut** [comment]
Thank you for the detailed update. Frankly, I am not sure what the best next step.
I would like to keep the interface and documentation across all file storages consistent.
However, some people might rely on the GCSDataStore- and FileStore-specific values in the event, so changing its type might break their application and I would like to avoid a breaking change, if possible.

Maybe it is possible to extend the objects from FileStore and GCSDataStore to also have the `id`, `upload_length` and `upload_metadata` properties. What do you think?

**Comment by Murderlon** [comment]
Closed in #344
